src/Common/src/System/Security/Cryptography/DerSequenceReader.cs

```diff
@@ -260,16 +284,28 @@ internal string ReadIA5String()
 
         internal DateTime ReadUtcTime()
         {
-            EatTag(DerTag.UTCTime);
+            return ReadTime(DerTag.UTCTime, "yyMMddHHmmss'Z'");
+        }
+
+        internal DateTime ReadGeneralizedTime()
+        {
+            // Currently only supports reading times with no fractional seconds or time differentials
+            // as RFC 2630 doesn't allow these. In case this is done, the format string has to be parsed
+            // to follow rules on X.680 and X.690.
+            return ReadTime(DerTag.GeneralizedTime, "yyyyMMddHHmmss'Z'");
+        }
+
+        private DateTime ReadTime(DerTag timeTag, string formatString)
```

Comment:
In the X509 
```C#
if (bytes.Length == 13)
{
    ...
}
```
Which hints that the object has no tag and length. Interop.Crypto.GetAsn1StringBytes(validityDatePtr) could be modified to return a the unencoded value, but it requires extra work as the inner OpenSSL structure stores the the tag and length as integers and would require re-encoding them or calling an appropriate i2d method, so probably it's better to leave this as is for now.